pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -356,9 +356,34 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+			if containerID, err := getContainerIDFromLegacyLogSymlink(logSymlink); err == nil {
```

Comment:
nit: This is fairly deeply nested.  You should consider reducing this by replacing:
```golang
if err := ...; err == nil {
        // do things
} else {
        klog
}
```
with 
```golang
err := ...
if err != nil {
        klog
        continue
}
// do things
```

Here and below.  It also moves error handling closer to the call that generates the error

Reply:
Looks like you did it below, but not on this line.

Reply:
It would be very rare that log filename for container cannot be parsed (I have relaxed the length for container Id by 64-6 = 58 characters).
If short container Id cannot be obtained, the file should be allowed to be GC'ed.
So switching the if / else wouldn't yield change in indentation.